_core/smart/clarityRef.md

```diff
@@ -0,0 +1,193 @@
+
+
+## Supported types
+
+This section lists the types available to smart contracts. The only atomic types supported by the Clarity are booleans, integers, fixed length buffers, and principals. 
+
+### Int type
+
+The integer type in the Clarity language is a 16-byte signed integer, which allows it to specify the maximum amount of microstacks spendable in a single Stacks transfer. The special `BlockHeightInt` you can obtain with the `get-block-info` function.
+
+### Bool type
+
+Public functions return a boolean result of `'true` or `'false`. If the function returns `'true`, then it is considered valid, and any changes made to the blockchain state are materialized. If the function returns `'false`, the function is considered invalid and does not effect the smart contract's state. 
+
+So, if the `foo.A` function calls `bar.B`, and `bar.B` returns `true`, but `foo.A` returns `'false`, no effects from calling `foo.A` materialize &mdash; including effects from `bar.B`. If, however, `bar.B` returns `'false` and `foo.A` returns `'true`, there may be some database effects that are materialized from `foo.A`, but no effects from calling `bar.B` materialize.
+
+### Buffer type
+
+Currently, you cannot create a `Buffer` type. They are used in returns, all of the hash functions:
+
+* `hash160`
+* `sha256`
+* `keccak256`
+
+The properties are all, themselves, buffers `time`, `header-hash`, `burnchain-header-hash`, and `vrf-seed`. 
+
+### List type
+
+Clarity supports lists of the atomic types. However, the only variable length lists in the language appear as function inputs; there is no support for list operations like append or join.
+
+### Principal type
+
+Clarity provides this primitive for checking whether or not the smart contract transaction was signed by a particular principal. Principals represent a spending entity and are roughly equivalent to a Stacks address. The principal's signature is not checked by the smart contract, but by the virtual machine. A smart contract function can use the  globally defined `tx-sender` variable to obtain the current principal.
+
+Smart contracts may also be principals (represented by the smart contract's identifier). However, there is no private key associated with the smart contract, and it cannot broadcast a signed transaction on the blockchain. A smart contract uses the special variable `contract-principal` to refer to its own principal.
+
+You can use the `is-contract?` to determine whether a given principal corresponds to a smart contract.
+
+### String type
+
+To create strings, you can use of the hash functions, `hash160`, `sha256`, and `keccak256`, for example:
+
+```cl
+(hash160 "aaron.id")
+```
+
+### Tuple type
+
+To support the use of named fields in keys and values, Clarity  allows the construction of named tuples using a function `(tuple ...)`, for example
+
+```cl
+(define imaginary-number-a (tuple #real 1 #i 2))
```

Comment:
The syntax for tuples no longer uses `#`s, so this example would be:

```
(define imaginary-number-a (tuple (real 1) (i 2)))
(define imaginary-number-b (tuple (real 2) (i 3)))
```

Reply:
Fixed.